.github/workflows/ci.yml

```
@@ -0,0 +1,29 @@
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+name: CI
+
+jobs:
+
+  build_and_test:
+    name: build and run tests
+    runs-on: ubuntu-latest
```

In the past, I spent a lot of time to fix suddenly broken CI jobs because the environment changes from time to time. Since this is very annoying, I strongly prefer to specify the exact environment. It also makes the CI configuration more expressive (who knows that `ubuntu-latest` currently isn't 20.04?).

```suggestion
    runs-on: ubuntu-18.04
```

---

I didn't even know that you could specify the exact version for the run target... :smile: It's also very well-hidden in the docs.

Yep, for C/C++ stuff it's definitely a good idea to use a fixed version.